TeamA/AuthorityUser/src/main/java/org/wso2/sample/is/sso/agent/SampleContextEventListener.java

```diff
@@ -0,0 +1,83 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.sample.is.sso.agent;
+
+import org.wso2.carbon.identity.sso.agent.SSOAgentConstants;
+import org.wso2.carbon.identity.sso.agent.bean.SSOAgentConfig;
+import org.wso2.carbon.identity.sso.agent.SSOAgentException;
+import org.wso2.carbon.identity.sso.agent.saml.SSOAgentX509Credential;
+import org.wso2.carbon.identity.sso.agent.saml.SSOAgentX509KeyStoreCredential;
+
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class SampleContextEventListener implements ServletContextListener {
+
+    private static Logger LOGGER = Logger.getLogger("org.wso2.sample.is.sso.agent");
+
+    private static Properties properties;
+
+    public void contextInitialized(ServletContextEvent servletContextEvent) {
+
+        properties = new Properties();
+        try {
+            if(servletContextEvent.getServletContext().getContextPath().contains("authorityuser")) {
+                properties.load(servletContextEvent.getServletContext().
+                        getResourceAsStream("/WEB-INF/classes/authorityuser.properties"));
+            } else if(servletContextEvent.getServletContext().getContextPath().contains("avis.com")) {
+                properties.load(servletContextEvent.getServletContext().
+                        getResourceAsStream("/WEB-INF/classes/avis.properties"));
+            }
+            InputStream keyStoreInputStream = servletContextEvent.getServletContext().
+                    getResourceAsStream("/WEB-INF/classes/wso2carbon.jks");
+            SSOAgentX509Credential credential =
+                    new SSOAgentX509KeyStoreCredential(keyStoreInputStream,
+                            properties.getProperty("KeyStorePassword").toCharArray(),
+                            properties.getProperty("IdPPublicCertAlias"),
+                            properties.getProperty("PrivateKeyAlias"),
+                            properties.getProperty("PrivateKeyPassword").toCharArray());
+            SSOAgentConfig config = new SSOAgentConfig();
+            config.initConfig(properties);
+            config.getSAML2().setSSOAgentX509Credential(credential);
+            config.getOpenId().setAttributesRequestor(new SampleAttributesRequestor());
+            servletContextEvent.getServletContext().
+                    setAttribute(SSOAgentConstants.CONFIG_BEAN_NAME, config);
+        } catch (IOException e){
+            LOGGER.log(Level.SEVERE, e.getMessage(), e);
+        } catch (SSOAgentException e) {
+            LOGGER.log(Level.SEVERE, e.getMessage(), e);
+        }
+    }
+
+    public void contextDestroyed(ServletContextEvent servletContextEvent) {
```

Comment:
Shall we throw `UnsupportedOperationException` if this method is not implemented?

Reply:
This is done through the following PR. 
https://github.com/wso2-incubator/qa-identity-solution/pull/21